### PR TITLE
Fixes 2319: add repo filtering by origin & type

### PR DESCRIPTION
**Pull request by jlsherrill**
## Summary

This adds two new attributes to repositories:  content_type (rpm), and origin (external, red_hat).

I adjusted the values of 'origin' from the original task as the only other 'origin' i can think of would be for 'uploaded' content.  In the future when we support uploading content, we want image builder to still be able to get only 'custom repos with external urls', so they could fetch repos with origin=external to get that.

## Testing steps

For dev:

After migrating, run this main.go file:
```
package main

import (
	"github.com/content-services/content-sources-backend/pkg/db"
	"github.com/content-services/content-sources-backend/pkg/seeds"
	"github.com/openlyinc/pointy"
)

func main() {
	db.Connect()
	d := db.DB
	seeds.SeedRepositoryConfigurations(d, 1, seeds.SeedOptions{OrgID: "1", ContentType: pointy.Pointer("rpm"), Origin: pointy.Pointer("external")})
	seeds.SeedRepositoryConfigurations(d, 1, seeds.SeedOptions{OrgID: "1", ContentType: pointy.Pointer("not-rpm"), Origin: pointy.Pointer("red_hat")})
}
```

Launch the server, with the feature disabled:
```
FEATURES_NEW_REPO_FILTERING_ENABLED=false go run cmd/content-sources/main.go  api
```

Fetch the default repo list:

```
curl -X GET --location "http://localhost:8000/api/content-sources/v1.0/repositories/" \
    -H "Content-Type: application/json" \
    -H "x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiIxIn0sImFjY291bnRfbnVtYmVyIjoiMSIsImludGVybmFsIjp7Im9yZ19pZCI6IjEifX19Cg=="
```

This should show only 1 repo, as its filtering by to rpm & external by default.

Launch the server with the feature enabled:
```
FEATURES_NEW_REPO_FILTERING_ENABLED=true go run cmd/content-sources/main.go  api
```

Fetch the default repo list:

```
curl -X GET --location "http://localhost:8000/api/content-sources/v1.0/repositories/" \
    -H "Content-Type: application/json" \
    -H "x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiIxIn0sImFjY291bnRfbnVtYmVyIjoiMSIsImludGVybmFsIjp7Im9yZ19pZCI6IjEifX19Cg=="
```

Now you should get both repos that were created as part of the main.go run

Try adding  origin=external  & content_type=rpm to the url:

```
curl -X GET --location "http://localhost:8000/api/content-sources/v1.0/repositories/?origin=external" \
    -H "Content-Type: application/json" \
    -H "x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiIxIn0sImFjY291bnRfbnVtYmVyIjoiMSIsImludGVybmFsIjp7Im9yZ19pZCI6IjEifX19Cg=="
```

```
curl -X GET --location "http://localhost:8000/api/content-sources/v1.0/repositories/?content_type=rpm" \
    -H "Content-Type: application/json" \
    -H "x-rh-identity: eyJpZGVudGl0eSI6eyJ0eXBlIjoiVXNlciIsInVzZXIiOnsidXNlcm5hbWUiOiIxIn0sImFjY291bnRfbnVtYmVyIjoiMSIsImludGVybmFsIjp7Im9yZ19pZCI6IjEifX19Cg=="
```

You should get expected results.
Other things to test:

1.  trying to update the content_type or origin via PUT or PATCH should be ignored
2.  specifying content_type or origin when creating a repo should be ignored.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2319

File: db/migrations/20230831162419_add_repo_fields.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+BEGIN;
+
+alter table repositories add column origin varchar default 'external' not null;
+alter table repositories add column content_type varchar default 'rpm' not null;
```
Comment (rverdile) [inline db/migrations/20230831162419_add_repo_fields.up.sql:4]: do you want to add "if not exists" to these"?

Comment (jlsherrill) [reply db/migrations/20230831162419_add_repo_fields.up.sql:4]: yes!

**Review by rverdile — COMMENTED** [review]
When making a PUT request, without specifying origin or content_type, would expect the values to be replaced with the default values? MapForUpdate in the model would need to be updated for that to work I think.

**Comment by jlsherrill** [comment]
No, these values should not be able to be updated at all.

**Review by rverdile — APPROVED** [review]
lgtm!

**Comment by jlsherrill** [comment]
@swadeley this actually needs a bindings update.   The added parameters are kinda useless right now and not really worthy of testing though.

**Comment by swadeley** [comment]
Hi

I have just deployed, not built API docs yet, ran API tests and noticed this:
`            # compare if repo introspection has happened in past 8 or 24 hours or not
>           assert repo_introspection_time >= max_last_introspection_time
E           assert datetime.datetime(2023, 9, 17, 16, 0, 7) >= datetime.datetime(2023, 9, 17, 16, 21, 36, 380229)
`

File: api/openapi.json [hunk 2]
```diff
@@ -1024,6 +1032,22 @@
...
+                    {
+                        "description": "Comma separated list of origins to filter (red_hat,external)",
+                        "in": "query",
+                        "name": "origin",
+                        "schema": {
+                            "type": "string"
+                        }
+                    },
+                    {
+                        "description": "content type of a repository to filter on (rpm)",
```
Comment (swadeley) [inline api/openapi.json:1045]:
```suggestion
                        "description": "Content type of a repository to filter on (rpm)",
```

**Comment by swadeley** [comment]
Hi, after building apidocs and installing the updated plugin, i get:
`TypeError: __init__() got an unexpected keyword argument 'origin'`

**Comment by swadeley** [comment]
Hi, my mistake. I had to add support QE code side to accept the two new strings: origin and content_type

**Pull request merged**